mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSLanguageRepository.kt

```diff
@@ -21,11 +21,15 @@ import jetbrains.mps.smodel.adapter.structure.MetaAdapterFactory
 import jetbrains.mps.smodel.language.ConceptRegistry
 import jetbrains.mps.smodel.language.LanguageRegistry
 import jetbrains.mps.smodel.runtime.illegal.IllegalConceptDescriptor
+import org.jetbrains.mps.openapi.language.SAbstractConcept
 import org.jetbrains.mps.openapi.module.SRepository
 import org.modelix.model.api.IConcept
 import org.modelix.model.api.ILanguageRepository
 
-class MPSLanguageRepository(private val repository: SRepository) : ILanguageRepository {
+data class MPSLanguageRepository(private val repository: SRepository) : ILanguageRepository {
+
+    fun getConcept(uid: String): SAbstractConcept? = (resolveConcept(uid) as? MPSConcept)?.concept
```

Comment:
If we change the return type of `MPSLanguageRepository.resolveConcept` to `MPSConcept?` then `getConcept` can be simplified to `resolveConcept(uid)?.concept`. An additional function is then probably not needed anymore. At least a  more distinctive name (`resolveMPSConcept`?) would be useful.

Reply:
fixed in [c6daac3](https://github.com/modelix/modelix.core/pull/487/commits/c6daac3191ab2b85a596b3725f6e4e1e265c47ac)